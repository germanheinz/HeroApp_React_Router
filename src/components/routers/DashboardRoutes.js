import React from 'react'
import { Route, Switch } from 'react-router'
import { DcScreen } from '../dc/DcScreen'
import { LoginScreen } from '../login/LoginScreen'
import { MarvelScreen } from '../marvel/MarvelScreen'
import { Navbar } from '../ui/Navbar'

export const DashboardRoutes = () => {
    return (
        <>

        <Navbar />
        <div>
            <Switch>
                {/* <Route exact path="/login" component={ LoginScreen } /> */}
                <Route exact path="/marvel" component={ MarvelScreen } />
                <Route exact path="/dc" component={ DcScreen } />
            </Switch>
        </div>
            
        </>
    )
}
