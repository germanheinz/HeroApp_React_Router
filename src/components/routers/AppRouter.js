import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route
  } from "react-router-dom";
import { LoginScreen } from '../login/LoginScreen';
import { Navbar } from '../ui/Navbar';
import { DashboardRoutes } from './DashboardRoutes';

export const AppRouter = () => {
    return (
     <Router>
      <div>
        <Switch>
           <Route exact path="/login" component={ LoginScreen } />
           <Route path="/" component={ DashboardRoutes } />
        </Switch>

      </div>
    </Router>
  );
}
