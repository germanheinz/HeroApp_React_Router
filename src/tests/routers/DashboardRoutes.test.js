import { DashboardRoutes } from "../../components/routers/DashboardRoutes"
import { mount } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { AuthContext } from "../../components/auth/AuthContext";
import { MemoryRouter } from "react-router";
 
Enzyme.configure({ adapter: new Adapter() });

describe('Testing en <DashboardRoutess />', () => {
    
    test('debe mostrarse correctamente', () => {

        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'german'
            }
        }
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <MemoryRouter>
                    <DashboardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>
        )

        console.log(wrapper.html())
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.text-info').text().trim()).toBe('german');


    })
})
