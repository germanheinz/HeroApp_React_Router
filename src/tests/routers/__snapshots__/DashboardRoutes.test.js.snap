// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing en <DashboardRoutess /> debe mostrarse correctamente 1`] = `ReactWrapper {}`;
